RadarSDK/RadarGeofence.m

```diff
@@ -127,52 +127,49 @@ - (instancetype _Nullable)initWithObject:(id)object {
         } else if ([type isEqualToString:@"polygon"] || [type isEqualToString:@"Polygon"] || [type isEqualToString:@"isochrone"]) {
             id geometryObj = dict[@"geometry"];
 
-            if (![geometryObj isKindOfClass:[NSDictionary class]]) {
-                return nil;
-            }
-
-            id coordinatesObj = ((NSDictionary *)geometryObj)[@"coordinates"];
-            if (![coordinatesObj isKindOfClass:[NSArray class]]) {
-                return nil;
-            }
-
-            NSArray *coordinatesArr = (NSArray *)coordinatesObj;
-            if (coordinatesArr.count != 1) {
-                return nil;
+            if ([geometryObj isKindOfClass:[NSDictionary class]]) {
```

Comment:
Hmmm, do you have an example of the new payload?

What we need to think through: presently the `geometry` object is where we maintain the coordinates of the `RadarGeofence` object. We're not maintaining it independently of the geometry for geofences (we _do_ maintain it independently for `RadarPlace`)

So either we need to somehow take the response and create a faux geometry where we keep the coordinates (perhaps defaulting to the above radius `0.0`) or we mirror the approach of `RadarPlace`. I think I prefer the former, maintaining a shell geometry — perhaps of a new type `Point`.

(A third option I don't think we'd opt for is having no coordinates on these geofences).

Reply:
As it is `RadarPolygonGeometry` has a center, radius and coordinates (which is nullable). I think we can just pass in a null into the coordinates field when it is nor being returned from the server.